Add service name template (#1813)
grafana/beyla: docs/sources/configure/metrics-traces-attributes.md

by this property.

Higher values reduce the load on the Kubernetes API service.

| YAML                      | Environment variable                 | Type     | Default |
|---------------------------|--------------------------------------|----------|---------|
| `service_name_template`   | `BEYLA_SERVICE_NAME_TEMPLATE`        | string   | (empty) |

> The service name can be templated with a go template, this makes it possible to create conditional or extended services names.

Reviewer: ```suggestion
You can template service name using Go templates, this makes it possible to create conditional or extended services names.
```

> The template has the following context available:

Reviewer: ```suggestion
The following context is available to the template:
```

```
Meta: (*informer.ObjectMeta)
  Name: (string)
  Namespace: (string)
  Labels:
    label1: lv1
    label2: lv2
  Annotations:
    Anno1: av1
    Anno2: av2
  Pod: (*PodInfo)
  ...

ContainerName: (string)
```

> The full object and structure can be found in the ```kubecache informer.pb.go```

Reviewer: ```suggestion
You can find the full object and structure in the `kubecache informer.pb.go` source file.
```

> An example value for a service name might look like this:

Reviewer: ```suggestion
Service name template examples:
```

```
{{- .Meta.Namespace }}/{{ index .Meta.Labels "app.kubernetes.io/name" }}/{{ index .Meta.Labels "app.kubernetes.io/component" -}}{{ if .ContainerName }}/{{ .ContainerName -}}{{ end -}}
```

or

```
{{- .Meta.Namespace }}/{{ index .Meta.Labels "app.kubernetes.io/name" }}/{{ index .Meta.Labels "app.kubernetes.io/component" -}}
```

> From the result only the first line is used, which is then trimmed to prevent white space in the service name.

Reviewer: ```suggestion
In this example, only the first line is used and trimmed to prevent white space in the service name.
```

by this property.

Higher values reduce the load on the Kubernetes API service.

| YAML                      | Environment variable                 | Type     | Default |
|---------------------------|--------------------------------------|----------|---------|
| `service_name_template`   | `BEYLA_SERVICE_NAME_TEMPLATE`        | string   | (empty) |

> The service name can be templated with a go template, this allows to create conditional or extended services names.

Reviewer 2: Our docs linter will have small suggestions like these.

The template has the following context available:

```
Meta: (*informer.ObjectMeta)
  Name: (string)
  Namespace: (string)
  Labels:
    label1: lv1
    label2: lv2
  Annotations:
    Anno1: av1
    Anno2: av2
  Pod: (*PodInfo)
  ...

ContainerName: (string)
```

> The full object and structure can be found in the kubecache informer.pb.go

Reviewer 2: There are few ways to workaround the linter for cases like these, make things capitalised or wrap them with back ticks ``.
